content/docs/user-guide/scripting/script-canvas/programmer-guide/behavior-context.md

```diff
@@ -4,91 +4,241 @@ description: Learn about the important relationship between Script Canvas and th
+The ability to bind an EBus to the behavior context enables scripting to become driven and modular. The two main use cases for reflecting EBuses to the behavior context are _event handlers_ and _events_.
 
-![Self EntityID](/images/user-guide/scripting/script-canvas/script-canvas-behavior-context-2.png)
+Events are typically defined in code as part of a `RequestBus`, and are generally handled by some code system such as a component. Event handlers are typically defined as part of a `NotificationBus`.
 
-The other bus that the Light component reflects to the behavior context is `LightComponentNotificationBus`. The following source code is also in the file `dev\Gems\LmbrCentral\Code\Source\Rendering\LightComponent.cpp`.
+In this example, we'll take a look at a simplistic **Light** component. The example shows how its behavior context reflection translates into Script Canvas nodes.
+
+In this Light component, the user can configure the light by setting parameters such as color, intensity, and radius. The Light component can also be turned on or off, and you can respond to these events when they occur. Communication with an entity's Light component is done through two EBuses: the `LightComponentRequestBus` and the `LightComponentNotificationBus`.
+
+### Request bus
+
+A request bus is an EBus that can send _events_. Events can be thought of as requests that are intended for a system or object to handle. Components can reflect their event methods to the behavior context to make them available to scripting environments such as Script Canvas.
+
+Here are a few of the C++ event methods in the Light component:
 
 ```cpp
-behaviorContext->EBus<LightComponentNotificationBus>("LightNotification", "LightComponentNotificationBus", "Notifications for the Light Components")
-    ->Attribute(AZ::Script::Attributes::Category, "Rendering")
-    ->Handler<BehaviorLightComponentNotificationBusHandler>();
+// Turn light on. Returns true if the light was successfully turned on.
+bool TurnOnLight();
+
+// Turn light off. Returns true if the light was successfully turned off.
+bool TurnOffLight();
+
+// Toggle light state.
+void ToggleLight();
 ```
 
-Notification buses are also known as *event handlers*. You can use these event handlers on a component to respond to the events that happen to the component. As part of reflection to the behavior context, the preceding code specifies that the `BehaviorLightComponentNotificationBusHandler` handles events for the Light component.
+These events are part of the `LightComponentRequestBus`. Their behavior is implemented by `LightComponent`.
+
+```cpp
+bool LightComponent::TurnOnLight()
+{
+    bool success = m_light.TurnOn();
+    if (success)
+    {
+        EBUS_EVENT_ID(GetEntityId(), LightComponentNotificationBus, LightTurnedOn);
```

Comment:
This style of invoking an EBUS is generally discouraged, this would change to:
`LightComponentNotificationBus::Event(GetEntityId(), &LightComponentNotifications::LightTurnedOn);`

Reply:
Ah, thanks! I pulled this in from legacy Light component code. Thanks for modernizing for me.